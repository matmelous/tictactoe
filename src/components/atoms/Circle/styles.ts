import styled from "styled-components";

export const CircleStyled = styled.div`
  position: relative;
  width: 100px;
  height: 100px;
  min-width: 100px;
  border-radius: 100px;
  transform: scale(0.6);
  box-shadow: inset 0px 0px 2px #8f2184, inset 0px 0px 1px #8f2184,
    0px 0px 10px #8f2184, 0px 0px 5px #8f2184;
  background-color: #eefcfc;
  position: relative;

  &::before {
    content: "";
    position: absolute;
    display: block;
    width: 90px;
    top: 5px;
    left: 5px;
    border-radius: 100px;
    height: 90px;
    background-color: #282c34;
    box-shadow: 0px 0px 2px #8f2184, 0px 0px 1px #8f2184,
      inset 0px 0px 10px #8f2184, inset 0px 0px 5px #8f2184;
  }
`;
