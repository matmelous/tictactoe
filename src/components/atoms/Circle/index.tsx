import React from "react";
import { CircleStyled } from "./styles";

const Circle = () => {
  return <CircleStyled role="circle" />;
};

export default Circle;
