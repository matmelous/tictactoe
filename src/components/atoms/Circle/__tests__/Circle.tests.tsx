import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import Circle from "../index";

describe("Circle", () => {
  afterEach(cleanup);

  it("renders", () => {
    const { asFragment } = render(<Circle />);
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByTestId("circle")).toBeInTheDocument();
  });
});
