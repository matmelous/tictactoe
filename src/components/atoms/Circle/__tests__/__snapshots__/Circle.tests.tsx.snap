// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle renders 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL wqqDO"
    data-testid="circle"
  />
</DocumentFragment>
`;
